main.go

```diff
@@ -1,15 +1,67 @@
 package main
 
 import(
-  "github.com/labstack/echo"
-  mw "github.com/labstack/echo/middleware"
+    "net/http"
+
+    "github.com/labstack/echo"
+    mw "github.com/labstack/echo/middleware"
 )
 
+type Group struct {
+    Id      int     `json:"id"`
+    Name    string  `json:"name"`
+}
+
+type ResponseTemplate struct {
+    Content    []Group    `json:"content"`
+    ErrorCode  int        `json:"error_code"`
+    ErrorText  string     `json:"error_text"`
+    StatusCode int        `json:"status_code"`
+    StatusText string     `json:"status_text"`
+    Success    bool       `json:"success"`
+}
+
 func main() {
-  e := echo.New()
-  e.Use(mw.Logger())
-  e.Use(mw.Recover())
+    e := echo.New()
+    
+    e.Use(mw.Logger())
+    e.Use(mw.Recover())
+
+    // Routes
+    e.Get("/groups", getAllGroups)
+    e.Get("/groups/:id/gifs", getGroupGifs)
+    e.Post("/groups", createGroup)
+    e.Post("/groups/:id/gifs", createGifInGroup)
+
+    e.Run(":1323")
```

Comment:
Generally good practice to bring the port into an environment variable.  Makes deploying the app easier, as most providers (like Heroku) dynamically generate this and you have to use their environment variable!

Check out https://golang.org/pkg/os/#Getenv